tests/test_babamul.rs

```diff
@@ -316,3 +328,509 @@ async fn test_babamul_filters_pixel_flags() {
+            let value = record_result.expect("Failed to read Avro record");
+
+            if let apache_avro::types::Value::Record(fields) = value {
+                // Ensure this record matches our object_id to avoid stale topic data
+                let has_object_id = fields.iter().any(|(name, val)| {
+                    name == "objectId"
+                        && matches!(val, apache_avro::types::Value::String(s) if s == &lsst_object_id)
+                });
+
+                if !has_object_id {
+                    continue;
+                }
+
+                if let Some((_, survey_matches_value)) =
+                    fields.iter().find(|(name, _)| name == "survey_matches")
+                {
+                    if let apache_avro::types::Value::Union(_, boxed) = survey_matches_value {
+                        if let apache_avro::types::Value::Record(fields) = &**boxed {
+                            if let Some((_, ztf_value)) =
+                                fields.iter().find(|(name, _)| name == "ztf")
+                            {
+                                if let apache_avro::types::Value::Union(_, ztf_boxed) = ztf_value {
+                                    if let apache_avro::types::Value::Record(obj_fields) =
+                                        &**ztf_boxed
+                                    {
+                                        found_match = obj_fields.iter().any(|(field_name, field_value)| {
+                                            field_name == "object_id"
+                                                && matches!(field_value, apache_avro::types::Value::String(s) if s == &ztf_match_id)
+                                        });
+                                        if found_match {
+                                            break 'outer;
+                                        }
+                                    }
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+
+    assert!(
+        found_match,
+        "Expected to find ZTF match with object_id: {} in Babamul message",
+        ztf_match_id
+    );
+
+    // Cleanup inserted fixtures to avoid leaking state between tests
+    ztf_aux_collection
+        .delete_many(doc! {"_id": {"$in": [&ztf_match_id]}})
+        .await
+        .expect("Failed to cleanup ZTF aux fixture after test");
+    lsst_alerts_collection
+        .delete_many(doc! {"_id": {"$in": [lsst_alert_id]}})
+        .await
+        .expect("Failed to cleanup LSST alerts fixture after test");
+    lsst_aux_collection
+        .delete_many(doc! {"_id": {"$in": [&lsst_object_id]}})
+        .await
+        .expect("Failed to cleanup LSST aux fixture after test");
+    lsst_cutouts_collection
+        .delete_many(doc! {"_id": {"$in": [lsst_alert_id]}})
+        .await
+        .expect("Failed to cleanup LSST cutouts fixture after test");
+}
```

Comment:
here, lots of collections are cleaned up. For the following test the cleanup isn't as extensive (only one collection is cleaned up). I wonder why these are treated differently, as the logic seems very similar otherwise?